### PR TITLE
Add GCP credential support

### PR DESCRIPTION
**Pull request by michaelgugino**
https://jira.coreos.com/browse/CLOUD-518

### CODE REVIEW

File: install/0000_30_machine-api-operator_00_credentials-request.yaml [hunk 1]
```diff
@@ -64,3 +64,22 @@ spec:
...
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: openshift-machine-api-gcp
+  namespace: openshift-cloud-credential-operator
+spec:
+  secretRef:
+    name: gcp-cloud-credentials
```
Comment (michaelgugino) [inline install/0000_30_machine-api-operator_00_credentials-request.yaml]: Matches installer's config: https://github.com/openshift/installer/pull/1937/files#diff-8363d0f158df5ef13f50d06bad08fa0cR77

File: install/0000_30_machine-api-operator_00_credentials-request.yaml [hunk 2]
```diff
@@ -64,3 +64,22 @@ spec:
...
+    namespace: openshift-machine-api
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: GCPProviderSpec
+    predefinedRoles:
```
Comment (michaelgugino) [inline install/0000_30_machine-api-operator_00_credentials-request.yaml]: https://github.com/openshift/cloud-credential-operator/pull/77/files#diff-40435189e1c826f8b404eb4c1d6baf81R31

File: install/0000_30_machine-api-operator_00_credentials-request.yaml [hunk 3]
```diff
@@ -64,3 +64,22 @@ spec:
...
+    - "compute.instanceAdmin.v1"
```
Comment (michaelgugino) [inline install/0000_30_machine-api-operator_00_credentials-request.yaml]: Upstream's roles:

https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/master/pkg/cloud/google/serviceaccount.go#L57

File: install/0000_30_machine-api-operator_00_credentials-request.yaml [hunk 4]
```diff
@@ -64,3 +64,22 @@ spec:
...
+    - "compute.securityAdmin"
+    - "iam.serviceAccountActor"
```
Comment (ingvagabund) [inline install/0000_30_machine-api-operator_00_credentials-request.yaml]: Do we need all the roles?

File: install/0000_30_machine-api-operator_00_credentials-request.yaml (hunk 3 above)
Comment (enxebre) [reply install/0000_30_machine-api-operator_00_credentials-request.yaml]: I doubt we need `compute.securityAdmin` and `iam.serviceAccountActor` is deprecated https://cloud.google.com/iam/docs/faq#in_what_scenarios_would_i_use_the_service_account_actor_role

**Pull request merged**